include/alpaka/queue/QueueCpuSync.hpp

```diff
@@ -186,8 +189,7 @@ namespace alpaka
                     queue::QueueCpuSync const & queue)
                 -> bool
                 {
-                    alpaka::ignore_unused(queue);
-                    return true;
+                    return !queue.m_spQueueImpl->m_bCurrentlyExecutingTask;
```

Comment:
I am wondering for which case this is useful? If I call enqueue, the task will not return until, so nobody except the task itself has a change to check to queue for emptyness. Of course a concurrent task could test the queue, but in that case `m_bCurrentlyExecutingTask `should be an atomic?

Reply:
The task being executed can be a host callback which can check the state (this is exactly what the unit test is doing).

Reply:
Ah, makes sense. I will just check the changes with my code. :wink: